torch/library.py

```diff
@@ -131,13 +137,10 @@ def impl(self, op_name, fn, dispatch_key=''):
         _impls.add(key)
         self._op_impls.add(key)
 
-    def __del__(self):
-        # _op_impls might not have been initialized if an error was thrown in __init__
-        _op_impls_ = getattr(self, '_op_impls', None)
-        if _op_impls_:
-            for key in self._op_impls:
-                _impls.remove(key)
-            del self.m
+
+def _del_library(captured_impls, op_impls):
+    for key in op_impls:
+        captured_impls.remove(key)
```

Comment:
These are both sets, right?
```suggestion
    captured_impls.difference_update(ops_impl)
```
or simply
```suggestion
    caputred_impls -= ops_impl
```

Reply:
Yes, you're correct